src/util/std_types.h

```diff
@@ -39,8 +31,8 @@ class bool_typet:public typet
   }
 };
 
-/*! \brief The NIL type
-*/
+/// The NIL type, i.e., an invalid type, no value. Use `optional<typet>`
+/// instead where possible.
 class nil_typet:public typet
```

Comment:
You might even want to deprecate this (though not sure you can deprecate a class, I think it would look like:
```C++
class DEPRECATED nil_typet : public typet
```

Reply:
Will do this in a follow-up PR